### PR TITLE
Added string caching feature for Speech Synthesis

### PR DESCRIPTION
**Pull request by anderm**
- Added caching feature for Speech Synthesis
- Fixed removed local audio map

### CODE REVIEW

**Comment by phongcao** [comment]
@anderm Looking great! Can you add a method to clear the cache in case we need to free up memory?

File: src/CognitiveServices/SpeechSynthesis.ts [hunk 1]
```diff
@@ -59,11 +60,31 @@ export class SpeechSynthesizer implements Speech.ISpeechSynthesizer {
...
+            // Replaces phonemes if it needs to, otherwise it returns the same text
+            text = this.replacePhonemes(text);
+        }
+
+        if (this._localAudioCacheMap && this._localAudioCacheMap.has(text)) {
+            // This text is already cached
+            return;
+        }
+
+        this.cacheSpeechData(text);
```
Comment (phongcao) [inline src/CognitiveServices/SpeechSynthesis.ts]: What will happen inside **cacheSpeechData** method if **this._localAudioCacheMap** is null?

Comment (anderm) [reply src/CognitiveServices/SpeechSynthesis.ts]: `this._localAudioCacheMap` is initialized in the constructor so it will never be null.

Comment (phongcao) [reply src/CognitiveServices/SpeechSynthesis.ts]: So we won't need to check for it in the if condition right?
`if (**this._localAudioCacheMap** && this._localAudioCacheMap.has(text))`

**Comment by anderm** [comment]
@phongcao - Good idea on clearing the memory - we would need to add some logic to check memory usage and clear after a threshold

**Comment by phongcao** [comment]
That's exactly right! It might be keep running until it's out of memory in production.

**Comment by anderm** [comment]
@phongcao added logic to clear the cache, please review

File: src/CognitiveServices/SpeechSynthesis.ts (hunk 1 above)
Comment (anderm) [reply src/CognitiveServices/SpeechSynthesis.ts]: That's right, I will remove that

**Pull request merged**